src/uu/rm/src/rm.rs

```diff
@@ -137,7 +144,19 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                 }
             },
             one_fs: matches.get_flag(OPT_ONE_FILE_SYSTEM),
-            preserve_root: !matches.get_flag(OPT_NO_PRESERVE_ROOT),
+            preserve_root: if matches.get_flag(OPT_NO_PRESERVE_ROOT) {
+                PreserveRoot::No
+            } else {
+                match matches
+                    .get_one::<String>(OPT_PRESERVE_ROOT)
+                    .unwrap()
+                    .as_str()
+                {
+                    "all" => PreserveRoot::YesAll,
+                    "default" => PreserveRoot::Default,
+                    _ => PreserveRoot::Default,
+                }
```

Comment:
Is this deliberately different from GNU `rm`? GNU `rm` only allows `all` and everything else leads to an `unrecognized --preserve-root-argument` error.

Reply:
hmm, good point :)